zerver/lib/export.py

```diff
@@ -0,0 +1,653 @@
+        if orig_dt is None:
+            continue
+        if timezone.is_naive(orig_dt):
+            logging.warning("Naive datetime:", item)
+            dt = timezone.make_aware(orig_dt)
+        else:
+            dt = orig_dt
+        utc_naive  = dt.replace(tzinfo=None) - dt.utcoffset()
+        item[field] = (utc_naive - datetime.datetime(1970, 1, 1)).total_seconds()
+
+# Export common, public information about the realm that we can share
+# with all realm users
+def export_realm_data(realm, response):
+    response['zerver_realm'] = [model_to_dict(x) for x in Realm.objects.filter(id=realm.id)]
+    floatify_datetime_fields(response, 'zerver_realm', 'date_created')
+
+    for (table, model) in realm_tables:
+        response[table] = [model_to_dict(x) for x in
+                           model.objects.filter(realm_id=realm.id)] # type: ignore # Iterating through types
+    response["zerver_client"] = [model_to_dict(x) for x in Client.objects.select_related()]
+
+# To export only some users, you can tweak the below UserProfile query
+# to give the target users, but then you should create any users not
+# being exported in a separate
+# response['zerver_userprofile_mirrordummy'] export so that
+# conversations with those users can still be exported.
+def export_with_admin_auth(realm, response, include_invite_only=True, include_private=True):
+    response['zerver_userprofile'] = [model_to_dict(x, exclude=["password", "api_key"])
+                                      for x in UserProfile.objects.filter(realm=realm)]
+    response['zerver_userprofile_crossrealm'] = [dict(email=x.email, id=x.id) for x in [
+        get_user_profile_by_email(settings.NOTIFICATION_BOT),
+        get_user_profile_by_email(settings.EMAIL_GATEWAY_BOT),
+        get_user_profile_by_email(settings.WELCOME_BOT),
+    ]]
+    floatify_datetime_fields(response, 'zerver_userprofile', 'date_joined')
+    floatify_datetime_fields(response, 'zerver_userprofile', 'last_login')
+    floatify_datetime_fields(response, 'zerver_userprofile', 'last_reminder')
+    user_profile_ids = set(userprofile["id"] for userprofile in response['zerver_userprofile'])
+    user_recipients = [model_to_dict(x)
+                       for x in Recipient.objects.filter(type=1)
+                       if x.type_id in user_profile_ids]
+    user_recipient_ids = set(x["id"] for x in user_recipients)
+    user_subscription_query = Subscription.objects.filter(user_profile__in=user_profile_ids,
+                                                          recipient_id__in=user_recipient_ids)
+    user_subscription_dicts = [model_to_dict(x) for x in user_subscription_query]
+
+    response["zerver_userpresence"] = [model_to_dict(x) for x in
+                                       UserPresence.objects.filter(user_profile__in=user_profile_ids)]
+    floatify_datetime_fields(response, 'zerver_userpresence', 'timestamp')
+
+    response["zerver_useractivity"] = [model_to_dict(x) for x in
```

Comment:
yeah, probably, since we do it like 20 times :).  I think I'll save that for #1567 (added to description there).